stetho-urlconnection/src/main/java/com/facebook/stetho/urlconnection/StethoURLConnectionManager.java

```diff
@@ -37,24 +31,37 @@
  */
 @NotThreadSafe
 public class StethoURLConnectionManager {
-  private static final AtomicInteger sSequenceNumberGenerator = new AtomicInteger(0);
+  private static boolean sIsStethoPresent = isStethoPresent();
+
+  private static boolean isStethoPresent() {
+    try {
+      Class.forName("com.facebook.stetho.Stetho");
+      return true;
+    } catch (ClassNotFoundException e) {
+      return false;
+    }
+  }
 
-  private final NetworkEventReporter mStethoHook = NetworkEventReporterImpl.get();
-  private final int mRequestId;
-  @Nullable private final String mFriendlyName;
+  private final Holder mHolder;
 
-  @Nullable private String mRequestIdString;
+  private static class Holder {
+    private final StethoURLConnectionManagerImpl impl;
 
-  private HttpURLConnection mConnection;
-  @Nullable private URLConnectionInspectorRequest mInspectorRequest;
+    public Holder(@Nullable String friendlyName) {
+      impl = new StethoURLConnectionManagerImpl(friendlyName);
+    }
+  }
 
   public StethoURLConnectionManager(@Nullable String friendlyName) {
-    mRequestId = sSequenceNumberGenerator.getAndIncrement();
-    mFriendlyName = friendlyName;
+    if (sIsStethoPresent) {
+      mHolder = new Holder(friendlyName);
+    } else {
+      mHolder = null;
+    }
   }
 
   public boolean isStethoEnabled() {
-    return mStethoHook.isEnabled();
+    return isStethoPresent() && mHolder.impl.isStethoActive();
```

Comment:
Ack'd.  I meant to use the member variable here but yes `mHolder` is the right call.